core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -55,8 +56,17 @@ private[spark] class ResultTask[T, U](
     // Deserialize the RDD and the func using the broadcast variables.
     val deserializeStartTime = System.currentTimeMillis()
     val ser = SparkEnv.get.closureSerializer.newInstance()
+    // Kryo deserialization is not thread-safe w.r.t. underlying buffer
+    // create a copy of the buffer if Kryo is being used
+    val copy = if (SparkEnv.get.closureSerializer.isInstanceOf[KryoSerializer]) {
```

Comment:
Here, you might be able to check whether `ser` is an instance of `KryoSerializerInstance` instead.